Bring /info API doc in line with the actual response.

draghuram opened the pull request:
The fix handles versions 1.18 and 1.19.

Closes #13139.

In addition to "Debug" and "IndexServerAddress" changes, I found that 1.18 doc didn't list the field "RegistryConfig" so I added it as well. 

In addition, 1.19 doc was not linked to from `mkdocs.yml` so I added that link. Please let me know if there is any problem with this.

thaJeztah commented:
Changes look good (thanks for the mkdocs, I think that has been an oversight when creating the 1.19 docs).

I don't have easy access to my test-machines at the moment, so if someone is able to verify the output :)

Thanks!

moxiegirl commented:
Can you update the "What's new" section

draghuram commented:
@moxiegirl Can you tell me where can I find this "What's new" section? Is it in `CHANGELOG.md`? Thanks.

HuKeping commented:
@draghuram  it's `docs/sources/reference/api/docker_remote_api.md` I think

draghuram commented:
@HuKeping Thanks.

@moxiegirl I updated "What's New" section.

moxiegirl commented:
Thanks @draghuram LGTM

ping @docker/docs-owners

thaJeztah commented:
Just checked the `/v1.18/info` output on a fresh host running 1.6.2 and it looks like there are some more fields that are output as an int, not a bool (`IPv4Forwarding`, `MemoryLimit`, `SwapLimit`). Also the keys in the output are alphabetically (but that's not directly related to this PR);

@draghuram could you verify this on your host as well and fix the other boolean fields in the 1.18 API?

```JSON
{
  "Containers": 0,
  "Debug": 0,
  "DockerRootDir": "/var/lib/docker",
  "Driver": "aufs",
  "DriverStatus": [
    [
      "Root Dir",
      "/var/lib/docker/aufs"
    ],
    [
      "Backing Filesystem",
      "extfs"
    ],
    [
      "Dirs",
      "0"
    ],
    [
      "Dirperm1 Supported",
      "true"
    ]
  ],
  "ExecutionDriver": "native-0.2",
  "ID": "JXGQ:7QYW:ESQH:54OR:TKXK:IXIX:B6F7:OVXD:5OS5:T2R5:U4LO:I3BD",
  "IPv4Forwarding": 1,
  "Images": 0,
  "IndexServerAddress": "https://index.docker.io/v1/",
  "InitPath": "/usr/bin/docker",
  "InitSha1": "7f9c6798b022e64f04d2aff8c75cbf38a2779493",
  "KernelVersion": "3.16.0-4-amd64",
  "Labels": null,
  "MemTotal": 1049055232,
  "MemoryLimit": 0,
  "NCPU": 1,
  "NEventsListener": 0,
  "NFd": 13,
  "NGoroutines": 20,
  "Name": "foobar",
  "OperatingSystem": "Debian GNU/Linux 8 (jessie)",
  "RegistryConfig": {
    "IndexConfigs": {
      "docker.io": {
        "Mirrors": null,
        "Name": "docker.io",
        "Official": true,
        "Secure": true
      }
    },
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ]
  },
  "SwapLimit": 0,
  "SystemTime": "2015-05-17T16:41:59.193875381-04:00"
}
```

draghuram commented:
Sure. I will check and update the PR, if required.

draghuram committed:
Bring /info API doc in line with the actual response.

The fix handles versions 1.18 and 1.19.

Closes #13139.

Signed-off-by: Raghuram Devarakonda <draghuram@gmail.com>

draghuram commented:
I updated the PR with the following changes:

- `IPv4Forwarding`, `MemoryLimit`, `SwapLimit` are updated to show int return in 1.18 doc as that is how 1.18 returns them.

- Added the new fields `CpuCfsPeriod`, `CpuCfsQuota`, and `OomKillDisable` to 1.19 info response as they seem to have been recently added. 

- /info response is now sorted by keys in both 1.18 and 1.19 docs.

- Updated "What's new' section.

I sincerely hope that the PR gets merged before any more new fields are added to the response :-).

thaJeztah commented:
> I sincerely hope that the PR gets merged before any more new fields are added to the response :-)

I'll save you that trouble. LGTM

Thanks so much!

draghuram commented:
Thanks. Much appreciated :-).
